### PR TITLE
locking a target should not keep it visible

### LINKED ISSUE #160: Locked targets remain locked even if outside of detection range
Hard to know if this is intended or not, but perhaps a case that does not practically present itself due to current balance.  It is certainly not what we expected, and may present issues for some new content coming later.

Currently, if a bot (A) that is invisible becomes visible (by deactivating a masker let's say) and is locked by another bot (B) if (A) reactivates their masker, or runs away, they will remain locked and visible to (B) until the target is lost to the locking range of (B), ecm, or willful unlocking.

Desired change:
A robot **locked or not**, will disappear from robots that can not see it (masking/detection).

Existing related behavior to remain unchanged:
A robot (A) invisible to another (B) will appear to (B) if (A) locks (B).

### PR DESCRIPTION
**Pull request by MikeJeffers**
Closes: #160

### CODE REVIEW

File: src/Perpetuum/Zones/Locking/LockValidator.cs [hunk 1]
```diff
@@ -31,6 +31,10 @@ public override void VisitUnitLock(UnitLock unitLock)
             {
                 Error = ErrorCodes.TargetOutOfRange;
             }
+            else if (!unitLock.Owner.IsVisible(unitLock.Target))
```
Comment (MikeJeffers) [inline src/Perpetuum/Zones/Locking/LockValidator.cs:34]: Actively removes target on lockHandler update

File: src/Perpetuum/Robots/Robot.cs [hunk 2]
```diff
@@ -353,14 +353,6 @@ public void CreateComponents()
             }
         }
 
-        protected override bool IsDetected(Unit target)
-        {
-            if (_lockHandler.IsLocked(target))
```
Comment (MikeJeffers) [inline src/Perpetuum/Robots/Robot.cs:358]: This was the logic that we seem to be at odds with.
Looks completely intentional... scary

**Comment by nref-dan** [comment]
Can we double check this doesn't negatively affect the NPC aggro system?

Also, as a balancing discussion I think it would be nice to see active targetting have a bonus to detection. i.e If I can only see you at 100m and lock you (assume my targetting range is 200m), you have to go out to 120m to drop the lock with masking.

**Comment by MikeJeffers** [comment]
> 
> 
> Can we double check this doesn't negatively affect the NPC aggro system?
> 
> Also, as a balancing discussion I think it would be nice to see active targetting have a bonus to detection. i.e If I can only see you at 100m and lock you (assume my targetting range is 200m), you have to go out to 120m to drop the lock with masking.

Interesting, locking has some drag into the masked distance..

As for NPC aggro, it should be the same as players in all cases.  NPC's account for visibility and locking range all the same as players.
The psuedo-aggro should actually handle the issue where npc's cant see an assist-aggressor (someone who does not target the npc directly, but should get aggro through assist modules).  This happens already.  But I am curious what cases you have in mind?

**Comment by nref-dan** [comment]
> 
> 
> > Can we double check this doesn't negatively affect the NPC aggro system?
> > Also, as a balancing discussion I think it would be nice to see active targetting have a bonus to detection. i.e If I can only see you at 100m and lock you (assume my targetting range is 200m), you have to go out to 120m to drop the lock with masking.
> 
> Interesting, locking has some drag into the masked distance..
> 
> As for NPC aggro, it should be the same as players in all cases. NPC's account for visibility and locking range all the same as players.
> The psuedo-aggro should actually handle the issue where npc's cant see an assist-aggressor (someone who does not target the npc directly, but should get aggro through assist modules). This happens already. But I am curious what cases you have in mind?

Literally what you just pointed out :) So long as it's accounted for that's fine, I just wanted to be sure we'd thought about it even if the answer is "it doesn't affect it"

**Comment by MikeJeffers** [comment]
This was overruled by design team, some pvp metas would be broken.
Oh well

**Pull request closed without merging**